app/routes/articles/ArticleDetailRoute.ts

```diff
@@ -36,6 +44,11 @@ const mapStateToProps = (state, props) => {
     });
   });
   const emojis = selectEmojis(state);
+  setTimeout(() => {
+    if (article?.slug && articleIdOrSlug !== article.slug) {
+      props.history.replace(`/articles/${article.slug}`);
+    }
+  }, 0);
```

Comment:
Did you try out `useHistory`?

This is quite hacky, but still better than nothing. I'd like a comment here to explain why the timeout is needed though.

Reply:
Read the rules of hooks (See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem).
useHistory cannot be used here.

I can add an explainer.

Reply:
Yes it wouldn't work in the route of course, but why not move it to the page component?

Reply:
Well, that might work just as well, I can do some experimentation with it

Reply:
Since it's just a `replace` I don't really mind where it is

Reply:
I've moved it for Joblistings and Articles, Events will have to stay like this until EventDetail is rewritten to a function component